src/components/Login/Login.tsx

```diff
@@ -0,0 +1,163 @@
+import { getCsrfToken, signIn } from "next-auth/react";
+import { useForm } from "react-hook-form";
+
+import { useState, useRef, useEffect } from "react";
+
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+  DialogFooter,
+  Switch,
+  Input,
+  Label,
+  Button,
+} from "~/components/ui";
+
+interface FormData {
+  email: string;
+  password: string;
+  type: string;
+}
+
+import { auth } from "~/pages/api/auth/[...nextauth]";
+import { sendPasswordResetEmail } from "firebase/auth";
+
+const Login = () => {
+  const [submitType, setSubmitType] = useState("sign-in");
+  const [formError, setFormError] = useState(" ");
+  const [wrongPassCount, setWrongPassCount] = useState(0);
+
+  const csrfInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    (async () => {
+      const csrfToken = await getCsrfToken();
+      if (csrfInput.current) {
+        csrfInput.current.value = csrfToken as string;
+      }
+    })();
+  });
```

Comment:
where is dependency array?

Reply:
so it's better to include an emoty one than not icnlude it at all?